selfdrive/car/isotp_parallel_query.py

```diff
@@ -123,21 +122,29 @@ def get_data(self, timeout, total_timeout=None):
         response_valid = dat[:len(expected_response)] == expected_response
 
         if response_valid:
-          last_response_time = time.monotonic()
+          response_timeouts[tx_addr] = time.monotonic() + timeout
+          pending_responses.discard(tx_addr)
           if counter + 1 < len(self.request):
             msg.send(self.request[counter + 1])
             request_counter[tx_addr] += 1
           else:
             results[tx_addr] = dat[len(expected_response):]
             request_done[tx_addr] = True
         else:
-          request_done[tx_addr] = True
-          cloudlog.warning(f"iso-tp query bad response: {tx_addr} - 0x{dat.hex()}")
+          error_code = dat[2] if len(dat) > 2 else -1
+          if error_code == 0x78:
+            response_timeouts[tx_addr] = time.monotonic() + self.response_pending_timeout
+            pending_responses.add(tx_addr)
+            if self.debug:
+              cloudlog.warning(f"iso-tp query response pending: {tx_addr}")
+          else:
+            request_done[tx_addr] = True
```

Comment:
Seems like you want to set the timeout to the current time any time you set an addr as done so you don't wait an extra timeout period below where it waits for the max timeout to be expired.

Reply:
A lot of the requests are of len 1, so it's possible for a brand that we miss some responses if we never increment the counter if we only get "done" responses. What's on the branch is similar to the previous behavior of incrementing the timeout for any response

Reply:
I am saying you break out of the loop when `cur_time - max(response_timeouts.values()) > 0` meaning after you get the last good response message you increment the timer for that rx addr and then wait for it to expire (but don't need to) but maybe this isn't worth worrying about.

Reply:
When everything is received we break anyway.

Reply:
yeah, so it only waits longer than it needs to when some requests do not complete (and those requests time out before the last successful request succeeds)

Reply:
Yeah. If we don't increment on a done response, what if we get a bunch of single frame responses? It would eventually timeout

Reply:
This is the current behavior too, so I think this is okay to merge. We can adjust the timeout in another PR